File: 989.Add-to-Array-Form-of-Integer.js
/**
 * @param {number[]} num
 * @param {number} k
 * @return {number[]}
 */
var addToArrayForm = function (num, k) {
  const result = [];
  let index = num.length - 1;
  let cary = 0;

  while (index >= 0 || k) {
    const v1 = k % 10;
    k = Math.floor(k / 10);
    const v2 = num[index] || 0;
    let sum = v1 + v2 + cary;

    if (sum > 9) {
      cary = Math.floor(sum / 10);
      sum = sum % 10;
    } else {
      cary = 0;
    }

    result.push(sum);

    index -= 1;
  }

  if (cary) {
    result.push(cary);
  }

  return result.reverse();
};

File: python/989.Add-to-Array-Form-of-Integer.py
class Solution:
    def addToArrayForm(self, num: List[int], k: int) -> List[int]:
        result = []
        cary = 0
        idx = len(num) - 1

        while idx >= 0 or k:
            num1 = num[idx] if idx >= 0 else 0
            num2 = k % 10
            k = k // 10

            total = num1 + num2 + cary

            cary = total // 10
            result.append(total % 10)
            idx -= 1

        if cary:
            result.append(cary)

        return result[::-1]